### PR TITLE
[23.1] Fix copying metadata to copied job outputs

### LINKED ISSUE #17003: Unavailable metadata after extracting non-terminal dataset from collection
**Describe the bug**
When running the `__EXTRACT_DATASET__` tool on a collection element, which hasn't reached the OK state yet, metadata, like index files, will not get copied to the extracted element.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: usegalaxy.eu; release_23.1_europe branch

**To Reproduce**
1. Run __EXTRACT_DATASET__ to extract a dataset, for which Galaxy will build an index (e.g. a bam file), from a collection while that dataset is not ready yet.
2. The __EXTRACT_DATASET__ job will enter the running state, but only finish once the collection element becomes available
3. The now green, extracted dataset will continue to say queued and the download button will start a download without presenting a choice to download the index. Downstream tools relying on the index will also fail.

@mvdbeek could be a side-effect of #16931 (which EU has installed) or a related bug that didn't get fixed by the PR.

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes https://github.com/galaxyproject/galaxy/issues/17003

(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/model/metadata.py [hunk 1]
```diff
@@ -667,7 +667,6 @@ def from_external_value(self, value, parent, path_rewriter=None):
                 # directory. Correct.
                 file_name = path_rewriter(file_name)
             mf.update_from_file(file_name)
-            os.unlink(file_name)
```
Comment (mvdbeek) [inline lib/galaxy/model/metadata.py:670]: This isn't the right place to do this, cleaning up the job working directory will handle this correctly. fixes the directory metadata strategy case

Comment (mvdbeek) [reply lib/galaxy/model/metadata.py:670]: I bet this is also what happened in https://github.com/galaxyproject/galaxy/pull/14087

**Comment by bgruening** [comment]
> FAILED test/integration/objectstore/test_remote_objectstore_cache_operations.py::TestCacheOperation::test_cache_repopulated_metadata_file - AssertionError: assert 3 == 2

That one might be related.

**Comment by mvdbeek** [comment]
> That one might be related.

yep, fixed in a rebase

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Merci!

**Pull request merged**